packages/building-webpack/modern-config.js

```diff
@@ -16,6 +20,27 @@ module.exports = userOptions => {
     ...userOptions,
   };
 
+  if (typeof options.input === 'string' && options.input.endsWith('index.html')) {
```

Comment:
What if we base this off the `indexHTML` property? So if `indexHTML` is set and `entry` is not set, we extract it from the `indexHTML` file?

Reply:
thought about it as well but that would be a breaking change... as it would result in different defaults...
```js
options = { indexHTML: 'foo.html' }

// result now
{ index:HTML: 'foo.html', entry: 'index.js' }
// result if we do change
{ index:HTML: 'foo.html', entry: '<what we find in indexHTML>' }
```

Reply:
This would work, right:

```
if (options.indexHTML && !options.entry) {
  options.entry = extractModulesFromHTML(options.indexHTML);
}
```

This way our config can have a great default, but also offer a more 'traditional' or 'custom' approach where needed.

Reply:
as discussed entry is always defined as we merge the default options before and we would like to align with buildling-rollup where it's called input as well